### ISSUE TITLE
Improve TrainEngines impl

### ISSUE THREAD

**Issue by marcinn** [MaSzyna-Reloaded/MaSzyna-API-wrapper#51]
For all TrainEngine* classes:

* improve enums names, make 
* make state keys consistent and self-descriptive
* check & fix defaults
* remove redundant prefixes from properties & method 
*  complete LoadFIZ_* impl

**Comment by JezSonic** [comment]
This class should contain:
- enum mapping `Maszyna::TEngineType` (Which is an enum class)
- TrainPowerSource enum currently held by TrainElectricEngine

(and probaly much more)

**Comment by JezSonic** [comment]
> consider moving direction_[increase|decrease]
I think we can leave that in the `TrainController` as for future cabin implementation, because directional switches in cabins are independent from each other. For example, if you change cabin in Impuls, leaving directional switch in `up` position in cabin A, in cabin B it would be e.g. in neutral position (or in the state you set it before leaving cabin B).

When you change cabin in ET22 or EU07, in real life you HAVE to set the directional switch to 0 otherwise the locomotive will ignore it and won't start or will go in wrong direction.

Moving it to `TrainEngine` class could be a problem as Cabin behavior also depends on that  (displaying information, setting elements properly etc.)

**Comment by JezSonic** [comment]
> move main_controller_[increase|decrease] from TrainController to TrainEngine

I think that it should be left along with `direction` in the `TrainController`. Hear me why:
Both of them are imo generic properties that would be used in many other parts/train systems, so having one generic reference in the `TrainController` would make it easier to manage everything else, 'cause as I have mentioned direction and for older locos like ET22, EU07 have small lamp in cabin that tells the driver whether he's on resistant position or not

[More information will be later in edit, I'm consulting some people right now]

**Comment by marcinn** [comment]
Yes. As I wrote earlier having direction in base controller class (or train engine, or other base for trainengine) could be enough. But please think about just one question - should the wagons have a specific direction and the possibility of changing it?

**Comment by JezSonic** [comment]
Technically I'm not sure but they have animated wheels so animation needs to be played properly (wheels should spin in the proper direction)

**Comment by marcinn** [comment]
Try with sm42 as an example: go forward at some kph, then switch direction to backward.

For movement you have velocity (the sign indicates it's direction).

Direction property is a for  "direction adjuster" (nastawnik kierunku jazdy)

**Comment by JezSonic** [comment]
Either way we should make some TrainController properties read-only to wagons/locos or even other sections if we need that (probably not, but some setters could just block setting property from wagon but allow from loco)

**Comment by marcinn** [comment]
Why do you want to implement workarounds instead of just designing just what we need?

As I wrote earlier we should talk about this, with renaming and decoupling behaviour logic from base classes in mind (and maybe separate cabin logic)

To sum up: You're right that TrainEngine may be not a good place for direction handling. We also know that base RailVehicle is too generic for direction handling. As we discussed in the other ticket, there will be "Locomotive" class and (AFAIR) something for "handcars". Maybe these classes will be a better place for direction handling, or we just introduce another class between RailVehicle and Locomotive/Handcar (ControllableRailVehicle ?)

**Comment by JezSonic** [comment]
Yeah that could solve our problem

**Comment by JezSonic** [comment]
Also, I don't wanna go with dozens of classes to instantiate each other.

I'm not sure if we would need another class like RailVehicle

We could just have a TrainController, NPCController and CarContoller or something but that's up to another discussion

**Comment by JezSonic** [comment]
Okay, so what's your proposition for name for
WWList? I have no ideas
